src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1525,6 +1527,25 @@ private function dumpValue($value, $interpolate = true)
             }
 
             return sprintf('array(%s)', implode(', ', $code));
+        } elseif ($value instanceof LocatorArgument) {
+            $services = $value->getValues();
+            $servicesMap = array();
+            $getterCode = 'function ($id) {'."\n";
+            foreach ($services as $k => $ref) {
+                $servicesMap[] = sprintf("'%s' => '%s'", $k, $ref);
+                $singleGetter = sprintf('            if (\'%s\' === $id || \'%s\' === $id) {', $k, $ref);
+                $singleGetter .= sprintf("\n               return %s;", $this->getServiceCall((string) $ref, $ref));
+                $singleGetter .= "\n            }\n";
+                $getterCode .= $singleGetter;
+            }
+            $getterCode .= '        }';
+
+            $hasserCode = 'function ($id) {'."\n";
+            $hasserCode .= sprintf('            $references = array(%s);', implode(', ', $servicesMap));
+            $hasserCode .= "\n\n".'            return isset($references[$id]) || in_array($id, $references, true);';
+            $hasserCode .= "\n        }";
```

Comment:
More a question of tastes and all, but instead of several concatenations, I would have used a nowdoc instead :

```php
<?php

// ...

$hasserCode = <<<'EOD'
$references = ...
EOD;
```

Not excluding a call to sprintf afterwards, of course